profiles/car.lua

```diff
@@ -390,6 +392,74 @@ function process_way(profile, way, result)
   }
 
   WayHandlers.run(profile,way,result,data,handlers)
+
+  -- now process relations data
+  local matched_refs = nil;
+  if result.ref then
+    local match_res = Relations.match_to_ref(relations, result.ref)
+
+    local ref = ''
+    for _, m in pairs(match_res) do
+      if ref ~= '' then
+        ref = ref .. '; '
+      end
+
+      if m.dir then
+        ref = ref .. m.ref .. ' $' .. m.dir
+      else
+        ref = ref .. m.ref
+      end
+    end
+
+    result.ref = ref
+  end
+
+end
+
+function process_relation(profile, relation, result)
+  local t = relation:get_value_by_key("type")
+
+  function add_extra_data(m)
+    local name = relation:get_value_by_key("name")
+    if name then
+      result[m]['route_name'] = name
+    end
+
+    local ref = relation:get_value_by_key("ref")
+    if ref then
+      result[m]['route_ref'] = ref
+    end
+  end
+
+  if t == 'route' then
+    local route = relation:get_value_by_key("route")
+    if route == 'road' then
+      for _, m in ipairs(relation:members()) do
+        -- process case, where directions set as role
+        local role = string.lower(m:role())
+        if role == 'north' or role == 'south' or role == 'west' or role == 'east' then
+          result[m]['route_direction'] = role
+          add_extra_data(m)
+        end
+      end
+    end
+
+    local direction = relation:get_value_by_key('direction')
+    if direction then
+      direction = string.lower(direction)
+      if direction == 'north' or direction == 'south' or direction == 'west' or direction == 'east' then
+        for _, m in ipairs(relation:members()) do
+          if m:role() == 'forward' then
```

Comment:
[This way](http://www.openstreetmap.org/way/261596707) ends up with a `ref` of `CA 17`, not `CA 17 $South` as expected, because its role in the [southbound CA 17 relation](http://www.openstreetmap.org/relation/309738) is empty. It’s pretty common for the role to be omitted on one-way roads (in this case, one side of a divided highway).